lib/template-parts.php

```diff
@@ -177,92 +170,103 @@ function filter_rest_wp_template_part_collection_params( $query_params ) {
-				$path_list = array();
-				if ( file_exists( $base_directory . '/block-template-parts' ) ) {
-					$nested_files      = new RecursiveIteratorIterator( new RecursiveDirectoryIterator( $base_directory . '/block-template-parts' ) );
-					$nested_html_files = new RegexIterator( $nested_files, '/^.+\.html$/i', RecursiveRegexIterator::GET_MATCH );
-					foreach ( $nested_html_files as $path => $file ) {
-						$path_list[] = $path;
-					}
-				}
-				return $path_list;
-			}
-
-			// Get file paths for all theme supplied template parts.
-			$template_part_files = get_template_part_paths( get_stylesheet_directory() );
-			if ( is_child_theme() ) {
-				$template_part_files = array_merge( $template_part_files, get_template_part_paths( get_template_directory() ) );
-			}
-			// Build and save each template part.
-			foreach ( $template_part_files as $template_part_file ) {
-				$content = file_get_contents( $template_part_file );
-				// Infer slug from filepath.
-				$slug = substr(
-					$template_part_file,
-					// Starting position of slug.
-					strpos( $template_part_file, 'block-template-parts/' ) + 21,
-					// Subtract ending '.html'.
-					-5
-				);
-				// Wrap content with the template part block, parse, and create auto-draft.
-				$template_part_string = '<!-- wp:template-part {"slug":"' . $slug . '","theme":"' . wp_get_theme()->get( 'TextDomain' ) . '"} -->' . $content . '<!-- /wp:template-part -->';
-				$template_part_block  = parse_blocks( $template_part_string )[0];
-				create_auto_draft_for_template_part_block( $template_part_block );
-			}
-		};
-
 		$args['meta_query'] = $meta_query;
 	}
 
 	return $args;
 }
 add_filter( 'rest_wp_template_part_query', 'filter_rest_wp_template_part_query', 99, 2 );
+
+/**
+ * Creates a template part auto-draft if it doesn't exist yet.
+ *
+ * @access private
+ *
+ * @param string $slug    Template part slug.
+ * @param string $theme   Template part theme.
+ * @param string $content Template part content.
+ */
+function create_auto_draft_for_template_part( $slug, $theme, $content ) {
+	// We check if an auto-draft was already created,
+	// before running the REST API calls
+	// because the site editor needs an existing auto-draft
+	// for each theme template part to work properly.
+	$template_part_query = new WP_Query(
+		array(
+			'post_type'      => 'wp_template_part',
+			'post_status'    => array( 'publish', 'auto-draft' ),
+			'title'          => $slug,
+			'meta_key'       => 'theme',
+			'meta_value'     => $theme,
+			'posts_per_page' => 1,
+			'no_found_rows'  => true,
+		)
+	);
+	$template_part_post  = $template_part_query->have_posts() ? $template_part_query->next_post() : null;
+	if ( ! $template_part_post ) {
+		wp_insert_post(
+			array(
+				'post_content' => $content,
+				'post_title'   => $slug,
+				'post_status'  => 'auto-draft',
+				'post_type'    => 'wp_template_part',
+				'post_name'    => $slug,
+				'meta_input'   => array(
+					'theme' => $theme,
```

Comment:
This post-meta is not used or needed for templates. I assume this was copied and pasted from the template part code and adding it here was unintentional?